src/payment/strategies/braintree/braintree-credit-card-payment-strategy.spec.ts

```diff
@@ -115,6 +121,60 @@ describe('BraintreeCreditCardPaymentStrategy', () => {
                 expect(error).toBeInstanceOf(MissingDataError);
             }
         });
+
+        it('initializes hosted form if feature is enabled and configuration is passed', async () => {
+            paymentMethodMock.config.isHostedFormEnabled = true;
+
+            const formOptions = {
+                fields: {
+                    cardName: { containerId: 'cardName' },
+                    cardNumber: { containerId: 'cardNumber' },
+                    cardExpiry: { containerId: 'cardExpiry' },
+                },
+            };
+
+            await braintreeCreditCardPaymentStrategy.initialize({
+                methodId: paymentMethodMock.id,
+                braintree: {
+                    form: formOptions,
+                },
+            });
+
+            expect(braintreePaymentProcessorMock.initializeHostedForm)
+                .toHaveBeenCalledWith(formOptions);
+        });
+
+        it('does not initialize hosted form if feature is not enabled', async () => {
+            paymentMethodMock.config.isHostedFormEnabled = false;
+
+            await braintreeCreditCardPaymentStrategy.initialize({
+                methodId: paymentMethodMock.id,
+                braintree: {
+                    form: {
+                        fields: {
+                            cardName: { containerId: 'cardName' },
+                            cardNumber: { containerId: 'cardNumber' },
+                            cardExpiry: { containerId: 'cardExpiry' },
+                        },
+                    },
+                },
+            });
+
+            expect(braintreePaymentProcessorMock.initializeHostedForm)
+                .not.toHaveBeenCalled();
+        });
+
+        it('does not initialize hosted form if configuration is not passed', async () => {
```

Comment:
do you need a test for `aymentMethodMock.config.isHostedFormEnabled = false;` but hosted form configuration is passed? This could already be covered, just double checking.

Reply:
`does not initialize hosted form if feature is not enabled` should cover that case.